src/main/cpp/main.cpp

```diff
@@ -178,6 +178,10 @@ int main(int argc, char *argv[]) {
 
   try {
     result = options.parse(argc, argv);
+    if (1 == argc) {
```

Comment:
* Instead of testing argc, the value of the positional element should be tested
* Should create a usage function so that the code is not duplicated from a few lines below.